x/stake/handler.go

```diff
@@ -178,62 +214,35 @@ func handleMsgDelegate(ctx sdk.Context, msg types.MsgDelegate, k keeper.Keeper)
 }
 
 func handleMsgBeginUnbonding(ctx sdk.Context, msg types.MsgBeginUnbonding, k keeper.Keeper) sdk.Result {
-	err := k.BeginUnbonding(ctx, msg.DelegatorAddr, msg.ValidatorAddr, msg.SharesAmount)
+	ubd, err := k.BeginUnbonding(ctx, msg.DelegatorAddr, msg.ValidatorAddr, msg.SharesAmount)
 	if err != nil {
 		return err.Result()
 	}
 
-	tags := sdk.NewTags(
-		tags.Action, tags.ActionBeginUnbonding,
-		tags.Delegator, []byte(msg.DelegatorAddr.String()),
-		tags.SrcValidator, []byte(msg.ValidatorAddr.String()),
-	)
-	return sdk.Result{Tags: tags}
-}
-
-func handleMsgCompleteUnbonding(ctx sdk.Context, msg types.MsgCompleteUnbonding, k keeper.Keeper) sdk.Result {
-
-	err := k.CompleteUnbonding(ctx, msg.DelegatorAddr, msg.ValidatorAddr)
-	if err != nil {
-		return err.Result()
-	}
+	finishTime := types.MsgCdc.MustMarshalBinary(ubd.MinTime)
 
 	tags := sdk.NewTags(
-		tags.Action, ActionCompleteUnbonding,
+		tags.Action, tags.ActionBeginUnbonding,
```

Comment:
Should we also tag the finish time?

Reply:
It gets returned as the Data field.  To we also want it in a tag?

Reply:
I think it would be convenient for applications.

Reply:
Imo, tags should be used as events to be listening for.  In what scenario would you *be listening for* all the unbondings that end at a certain time?

Reply:
That seems unlikely, but I think you might well want to *search for* all unbondings ending in a time range, which also requires tags.

Reply:
That can be done with a Querier, that reads from the unbonding queue in state.

Reply:
is there the capability currently to query tag ranges? if not, this doesn't sound like an application of tags - I think of tags as categories not labels. - I guess it's kind of arbitrary though

Reply:
Hmm according to Tendermint docs there is, e.g. https://github.com/tendermint/tendermint/blob/master/state/txindex/kv/kv_test.go#L77

Reply:
Okay - let's do it. I've been converted. I'd recommend that (in the future) we refine our tagging system to more clearly distinguish between "categorical" vs "range" values

Reply:
Hmm, I still disagree with this, because tags add additional bloat to the chain, which queriers that just read state do not.  However, I'll add it until I get time to add a proper querier for this.